core/application/bloodDonationWorkflow/BloodDonationService.ts

```diff
@@ -123,49 +135,53 @@ export class BloodDonationService {
   async updateBloodDonation(
     donationAttributes: UpdateBloodDonationAttributes,
     bloodDonationRepository: BloodDonationRepository<DonationDTO>
-  ): Promise<string> {
-    try {
-      const { seekerId, requestPostId, donationDateTime, createdAt, ...restAttributes } =
-        donationAttributes
-      const item = await bloodDonationRepository.getDonationRequest(
-        seekerId,
-        requestPostId,
-        createdAt
-      )
+  ): Promise<BloodDonationResponseAttributes> {
+    const { seekerId, requestPostId, donationDateTime, createdAt, ...restAttributes } =
+      donationAttributes
+    const item = await bloodDonationRepository.getDonationRequest(
+      seekerId,
+      requestPostId,
+      createdAt
+    )
 
-      if (item === null) {
-        throw new Error('Item not found.')
-      }
+    if (item === null) {
+      throw new BloodDonationOperationError('Item not found.', GENERIC_CODES.ERROR)
```

Comment:
Use NOT_FOUND here.

Reply:
Updated in f98450b71743d5473aa397d4854e7347002cc21c